src/app/_hooks/useCollapseRange.ts

```diff
@@ -0,0 +1,85 @@
+import { useMemo } from 'react'
+
+// Inspiration: https://atlassian.design/components/pagination
+
+const DISTANCE_FROM_START = 3
+const DISTANCE_FROM_END = 2
+const ELLIPSIS = '...'
+
+export function useCollapseRange(
+  pageCount: number,
+  currentPage: number,
+  range: number = 7,
+): Array<number | typeof ELLIPSIS> {
+  const pageNumbers = useMemo(() => {
+    return Array.from({ length: pageCount }, (_, i) => i + 1)
+  }, [pageCount])
+
+  // Boolean
+  const needEllipsis = pageNumbers.length > range
+  const needsStartEllipsis = currentPage - DISTANCE_FROM_START > 0
+  const needsEndEllipsis = currentPage + DISTANCE_FROM_END < pageNumbers.length
+
+  // page numbers
+  const firstPageNumber = pageNumbers[0]
+  const lastPageNumber = pageNumbers[pageNumbers.length - 1]
+
+  // Indexes of page numbers
+  const firstPageNumberIndex = pageNumbers.indexOf(firstPageNumber)
+  const currentPageIndex = pageNumbers.indexOf(currentPage)
+
+  // Early returns
+  if (range <= 0) throw new Error('range must be greater than 0')
+  if (!needEllipsis) return pageNumbers
+  if (range === 1) return [currentPage]
+
+  // If range is less than 4, we don't show any ellipsis
```

Comment:
Instead of having comments, could we abstract this into a function with appropriate naming?